daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -151,27 +151,31 @@ object TransactionCoder {
         }
 
       case e: NodeExercises[Nid, Cid, Val] =>
-        encodeVal(e.chosenValue).map {
-          case (vversion, arg) =>
-            val exBuilder =
-              TransactionOuterClass.NodeExercise
-                .newBuilder()
-                .setChoice(e.choiceId)
-                .setTemplateId(ValueCoder.encodeIdentifier(e.templateId, Some(vversion))._2)
-                .setChosenValue(arg)
-                .setConsuming(e.consuming)
-                .setContractIdOrStruct(encodeCid, transactionVersion, e.targetCoid)(
-                  _.setContractId(_),
-                  _.setContractIdStruct(_))
-                .addAllActors(e.actingParties.map(_.underlyingString).asJava)
-                .addAllChildren(e.children.map(encodeNid).toList.asJava)
-                .addAllControllers(e.controllers.map(_.underlyingString).asJava)
-                .addAllSignatories(e.signatories.map(_.underlyingString).asJava)
-                .addAllStakeholders(e.stakeholders.map(_.underlyingString).asJava)
+        for {
+          retValue <- encodeVal(e.exerciseResult)
```

Comment:
We similarly guard with `transactionVersion precedes` in encoding.

Reply:
Done